test/cpp/api/dataloader.cpp

```diff
@@ -97,17 +97,19 @@ TEST(DataTest, ChunkDataSetWithInvalidInitParameter) {
 
   auto initialization_function =
       [&](size_t preloader_count, size_t batch_size, size_t cache_size) {
+        std::shared_ptr<datasets::detail::ChunkSelector> chunk_selector =
```

Comment:
If ChunkSelector is going to be a public class, should we move it out of detail namespace?